noir-projects/noir-contracts/contracts/app_subscription_contract/src/main.nr

```diff
@@ -102,8 +86,13 @@ contract AppSubscription {
             nonce
         ).call(&mut context);
 
-        // Assert that the given expiry_block_number < current_block_number + SUBSCRIPTION_DURATION_IN_BLOCKS.
-        AppSubscription::at(context.this_address()).assert_block_number(expiry_block_number).enqueue_view(&mut context);
+        // Assert that the `expiry_block_number - SUBSCRIPTION_DURATION_IN_BLOCKS < current_block_number`.
+        // --> We do that via the router contract to conceal which contract is performing the check.
+        Router::at(ROUTER_ADDRESS).check_block_number(
+            expiry_block_number - SUBSCRIPTION_DURATION_IN_BLOCKS,
+            Comparator.LT
```

Comment:
Shouldn't this be GT?

Reply:
Oh I see, lhs and rhs are swapped in the router. I'd change that as this is very confusing otherwise.

Reply:
You have a point. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/8384